### ISSUE TITLE
parallel operations?

### ISSUE THREAD

**Issue by natebrennand** [natebrennand/cpp_collections#6]
Some functional languages have these cool map methods that put multiple threads to work on collection operations.
See: https://clojuredocs.org/clojure.core/pmap

I think this would be immensely interesting to implement and also very useful.

Thoughts? @ianzapolsky @udaysinghcode

**Comment by natebrennand** [comment]
Also see [tesser](https://github.com/aphyr/tesser) for ideas of how we could do fold

**Comment by ianzapolsky** [comment]
multiple threads would be dope

**Comment by ianzapolsky** [comment]
doing research now

**Comment by natebrennand** [comment]
I'm not yet sure how we'd work out the number of threads to use yet. You generally want to configure that but idk if you would change a setting for the library or pass in a # when calling the fn.

**Comment by udaysinghcode** [comment]
Oh this is pretty cool - yeah I don't know how many threads you need to break it into to do a concurrent fold but the idea seems cool.

**Comment by ianzapolsky** [comment]
I'll take a crack at implementing concurrent map. Any ideas on how we should allow the user to specify the number of threads. Should he be limited by some multiple of the size of the collection? Or should we just take care of that calculation?

**Comment by natebrennand** [comment]
How would we calculate that without knowing how CPU or IO intensive the function is?
I think letting the user specify is the best move for now.

**Comment by ianzapolsky** [comment]
i'm having weird compiler issues with this stuff. Just trying a super simple multithread op. as a member function (printing from a thread), gives me this:
```
g++ -std=c++11 pass_conmap.cpp
Undefined symbols for architecture x86_64:
  "Collection<int>::bar()", referenced from:
      _main in pass_conmap-0c64ab.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
any idea @natebrennand ?

**Comment by natebrennand** [comment]
closed by #30
